Library/Homebrew/utils/bottles.rb

```diff
@@ -8,6 +8,8 @@ module Utils
   #
   # @api private
   module Bottles
+    EXTNAME_REGEX = /(\.([a-z0-9_]+)\.bottle\.((\d+)\.)?tar\.gz)$/.freeze
```

Comment:
```suggestion
    EXTNAME_REGEX = /\.([a-z0-9_]+)\.bottle\.((\d+)\.)?tar\.gz$/.freeze
```
Is this simpler? You'll need to stop throwing away the first entry of `EXTNAME_REGEX.match(filename).to_a` though.

Also, named captures might make `extname_tag_rebuild` a bit more readable.